modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/APITagVisibilityByRoleTestCase.java

```diff
@@ -149,13 +149,33 @@ public void testAPITagVisibilityAnonymousUser() throws Exception {
     @Test(groups = {"wso2.am"}, description = "Test the API tag visibility as a authorised user",
             dependsOnMethods = "testAPITagVisibilityAnonymousUser")
     public void testAPITagVisibilityAuthorisedUser() throws Exception {
-        List<TagDTO> tagDto = apiStoreClientAllowedUser.getAllTags().getList();
+        List<TagDTO> tagDtoList = apiStoreClientAllowedUser.getAllTags().getList();
         List<String> tagList = new ArrayList<>();
-        tagDto.forEach((tempDto) -> {
-            tagList.add(tempDto.getValue());
-        });
+        for (TagDTO tempTagDTO : tagDtoList) {
+            tagList.add(tempTagDTO.getValue());
+        }
         Assert.assertTrue(tagList.contains(TAGS_PUBLIC_API),
                 "Public visibility tag is not available for authorised user");
+        int retry = 10;
+        if (!tagList.contains(TAGS_RESTRICTED_API)) {
+            Thread.sleep(3000);
+            for (int i = 0; i < retry; i++) {
+                tagDtoList = apiStoreClientAllowedUser.getAllTags().getList();
+                tagList = new ArrayList<>();
+                for (TagDTO tempTagDTO : tagDtoList) {
+                    tagList.add(tempTagDTO.getValue());
+                }
+
+                if (!tagList.contains(TAGS_RESTRICTED_API)) {
+                    retry++;
```

Comment:
Why are we incrementing the retry value here?

Reply:
That was a mistake. Thanks for pointing it out. :)

Reply:
That was a mistake. Thanks for pointing it out. :) Fixed with b6fbaee449a6816702f0d1d639812630f02bf348